packages/grid/_modules_/grid/hooks/features/pagination/useGridPageSize.ts

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { GridApiRef } from '../../../models';
+import { GridComponentProps } from '../../../GridComponentProps';
+import { GridPageSizeApi } from '../../../models/api/gridPageSizeApi';
+import { useGridApiMethod } from '../../root';
+import { GRID_PAGE_SIZE_CHANGE } from '../../../constants';
+import { useLogger } from '../../utils';
+import { useGridSelector, useGridState } from '../core';
+import { visibleGridRowCountSelector } from '../filter';
+import { gridContainerSizesSelector } from '../../root/gridContainerSizesSelector';
+
+export const useGridPageSize = (
+  apiRef: GridApiRef,
+  props: Pick<GridComponentProps, 'pageSize' | 'onPageSizeChange' | 'autoPageSize'>,
+) => {
+  const logger = useLogger('useGridPageSize');
+  const [, setGridState, forceUpdate] = useGridState(apiRef);
+  const visibleRowCount = useGridSelector(apiRef, visibleGridRowCountSelector);
+  const containerSizes = useGridSelector(apiRef, gridContainerSizesSelector);
+
+  const setPageSize = React.useCallback(
+    (pageSize: number) => {
+      logger.debug(`Setting page size to ${pageSize}`);
+
+      setGridState((state) => ({
+        ...state,
+        pageSize,
+      }));
+      forceUpdate();
+    },
+    [setGridState, forceUpdate, logger],
+  );
+
+  React.useEffect(() => {
+    apiRef.current.updateControlState({
+      stateId: 'pageSize',
+      propModel: props.pageSize,
+      propOnChange: props.onPageSizeChange,
+      stateSelector: (state) => state.pageSize,
```

Comment:
this can be in the pagination state as well.
We can have 2 hooks working on the same substate in the case of pagination I think

Reply:
I was trying to be as close as possible from 1 hook = 1 substate
I suppose we have 2 approaches :

1) We have a single sub state `state.pagination` used by both hooks
2) We have two sub state `state.page` and `state.pageSize`.

Reply:
Reason for 1. would be, as these 2 hooks serve the same feature family then they can share the same substate I think. 

It could also be flat  as 2 . but then other prop would also need to be in the state such as pageCount.  Therefore, if you keep it flat you might pollute the first level of the state by adding your props directly, there which would impact the state  shape visibility... Therefore I lean towards 1.